### ISSUE TITLE
Usage of the champion static data api is very slow and causes all of the pages I have that use it to lag considerably.

### ISSUE THREAD

**Issue by zachcd** [robrua/cassiopeia#49]
Currently I have a page that calls the a match object to get champion images for champions played that match.  I believe this call ends up getting all champion information instead of just the image link I request.  I believe this is causing tremendous lag on the page and I'm sure that's not intended.  There's nowhere in the documentation that I can see where I can turn this type of requesting off.

**Comment by robrua** [comment]
Can you be a bit more specific about what you're doing, including what load policy and data store you're using?

**Comment by zachcd** [comment]
I've tried various load policies, I don't know what you mean by data stores in the documentation.

b1 = "http://ddragon.leagueoflegends.com/cdn/6.1.1/img/champion/" + game.blue_team.bans[0].champion.image.link
			b2 = "http://ddragon.leagueoflegends.com/cdn/6.1.1/img/champion/" + game.blue_team.bans[1].champion.image.link
			b3 = "http://ddragon.leagueoflegends.com/cdn/6.1.1/img/champion/" + game.blue_team.bans[2].champion.image.link
			p1 = "http://ddragon.leagueoflegends.com/cdn/6.1.1/img/champion/" + game.red_team.participants[0].champion.image.link
			p2 = "http://ddragon.leagueoflegends.com/cdn/6.1.1/img/champion/" + game.red_team.participants[1].champion.image.link
			p3 = "http://ddragon.leagueoflegends.com/cdn/6.1.1/img/champion/" + game.red_team.participants[2].champion.image.link
			p4 = "http://ddragon.leagueoflegends.com/cdn/6.1.1/img/champion/" + game.red_team.participants[3].champion.image.link
			p5 = "http://ddragon.leagueoflegends.com/cdn/6.1.1/img/champion/" + game.red_team.participants[4].champion.image.link

Basically this lags the whole page.

**Comment by jjmaldonis** [comment]
Is this data being pulled from a database via sqlalchemy or from Riot's servers? If it's from a database, which type are you using?

**Comment by robrua** [comment]
If you haven't set a data store you're using the in-memory cache by default. I could imagine some problems w/lazy loading if you hadn't already cached all the champions, as Cass would be making a new static data request for each of these lines. If you're using eager, it shouldn't be making another request after the first (maybe not even there). Are you sure the problem isn't with how you're loading the images themselves?

**Comment by zachcd** [comment]
I have a print statement that prints before and after this code and almost all of the delay is between them.  The actual access of images is done browser side.  I believe the lag only occurs when the data hasn't already been cached, as when accessing the page a second time, it loads very quickly.

**Comment by robrua** [comment]
doesn't sound unreasonable if your connection is slow and you're using eager loading. If you call get_champions() when you initialize your app it'll pre-cache the champion data and you won't have to grab it on the first page load. You can couple this w/lazy loading if you want it as fast as possible.

**Comment by zachcd** [comment]
oh perfect, get_champions() will probably solve this problem.  I'm still surprised as this is a high load webserver provided by amazon, I don't think it's limited by speeds, but I could be wrong... Thank you for your help.

**Comment by robrua** [comment]
No problem - let me know if that solves your issue.

**Comment by zachcd** [comment]
Absolutely!  That did work!  Looks like I just needed to preload all the data.
